consensus/src/threshold_simplex/actors/voter/actor.rs

```diff
@@ -369,40 +369,24 @@ impl<
             return None;
         }
         let proposal = self.proposal.as_ref().unwrap().clone();
-
-        // Ensure we have a notarization
-        let Some(notarization) = &self.notarization else {
-            return None;
-        };
-        let seed_signature = notarization.seed_signature;
-
-        // Check notarization and finalization proposal match
-        if notarization.proposal != proposal {
```

Comment:
While unnecessary, wondering if we should still perform this check as a sanity case?

Reply:
I removed the check because there was nothing to check against (since finalization.proposal is built from proposal). I re-added it again now that we check if a notarization exists (to extract the seed signature from it.)